flang/docs/OpenMPStandardsSupport.md

```diff
@@ -0,0 +1,62 @@
+<!--===- docs/FortranStandardsSupport.md
+
+   Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+   See https://llvm.org/LICENSE.txt for license information.
+   SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+-->
+
+# Flang OpenMP Standards Support
+
+```{contents}
+---
+local:
+---
+```
+
+This document summarizes OpenMP standards support in Flang. The information is only provided as a guideline. The
+TODOs/Not Yet Implemented messages emitted by the compiler for unimplemented features should be treated as authoritative.
+Standards support is provided upto OpenMP 4.0 for now. It will be extended later for OpenMP 4.5, OpenMP 5.* and OpenMP 6.0.
+
+The standards support information is provided as a table with three columns that are self explanatory. The Status column uses
+the letters **P**, **Y**, **N** for the implementation status:
+- **P** : When the implementation is incomplete for a few cases
+- **Y** : When the implementation is complete
+- **N** : When the implementation is absent
+
+Note : No distinction is made between the support in the Parser/Semantics, MLIR or Lowering support, and OpenMPIRBuilder support.
+
+## OpenMP 4.0
+
+| Feature                                                    | Status | Comments                                                |
+|------------------------------------------------------------|--------|---------------------------------------------------------|
+| proc_bind clause                                           | Y      | |
+| simd construct                                             | P      | Some clauses are not supported |
+| declare simd construct                                     | N      | |
+| do simd construct                                          | Y      | |
+| target data construct                                      | P      | |
+| target construct                                           | P      | |
+| target update construct                                    | P      | |
+| declare target directive                                   | P      | |
+| teams construct                                            | P      | |
+| distribute construct                                       | P      | |
+| distribute simd construct                                  | P      | |
+| distribute parallel loop construct                         | P      | |
+| distribute parallel loop simd construct                    | P      | |
+| depend clause                                              | P      | Depend clause with array sections are not supported |
+| declare reduction construct                                | N      | |
+| atomic construct extensions                                | Y      | |
```

Comment:
I don't think there is any support for atomic: seq_cst yet

Reply:
`atomic-16.f90` in the gfortran testsuite doesn't parse and looks like it is supposed to work. Surprisingly I don't find any tests for this feature in the Fujitsu testsuite.

Reply:
The parse error is probably the result of the `end atomic` statement becoming optional for the capture directive in OpenMP 5.0 or later.